automated-record-card/index.html

```diff
@@ -0,0 +1,196 @@
+        inputRecordID.addEventListener('input', keyboardInputHandler);
+        inputRecordID.addEventListener('blur', validateInput);
+        window.addEventListener('onbeforeunload', unloadHandler);
+
+        function validateInput() {
+          if (valueRecordID !== undefined && valueRecordID !== inputRecordID.value) {
+            document.getElementById('find-record').click();
+          }
+        }
+
+        function valueChangeHandler (value) {
+          inputRecordID.value = value || '';
+        }
+
+        function keyboardInputHandler () {
+          const value = inputRecordID.value;
+          if (typeof value !== 'string' || value === '') {
+            fieldRecordID.removeValue();
+          } else {
+            fieldRecordID.setValue(value);
+          }
+        }
+
+        function unloadHandler () {
+          window.removeEventListener('onbeforeunload', unloadHandler);
+          inputRecordID.removeEventListener('input', keyboardInputHandler);
+          detachValueChangeHandler();
+        }
+
+        // find a value in one of the defined languages
+        function languageValue (value) {
+          const languageKeys = ['en', 'eng', 'def', 'und'];
+          for (let key of languageKeys) {
+            if (value[key]) {
+              return value[key][0];
+            }
+          }
+          return;
+        }
+
+        // clear all fields
+        function clearEntry () {
+          // clear message
+          document.getElementById('message').innerText = '';
+
+          // clear fields
+          extension.entry.fields.name.removeValue();
+          extension.entry.fields.creator.removeValue();
+          extension.entry.fields.provider.removeValue();
+          extension.entry.fields.thumbnailUrl.removeValue();
+
+          document.getElementById('thumbnail-preview').innerHTML = '';
+        }
+
+        // click reset button
+        document.getElementById('reset-record').addEventListener('click', e => {
+          e.preventDefault();
+          clearEntry();
+          // clear identifier
+          extension.entry.fields.identifier.removeValue();
+          valueRecordID = extension.entry.fields.identifier.getValue();
+        });
+
+        // click go button
+        document.getElementById('find-record').addEventListener('click', e => {
+          e.preventDefault();
+
+          // empty fields
+          clearEntry();
+
+          // message
+          document.getElementById('message').innerText = 'Processing';
+
+          // is there a value
+          if (inputRecordID.value === '') {
+            document.getElementById('message').innerText = 'Add a record!';
+            return;
+          }
+
+          const recordId = inputRecordID.value.match('\/[0-9]+\/[a-zA-Z0-9_]+$')[0];
+          const xhr = new XMLHttpRequest();
+
+          xhr.open('GET', 'https://www.europeana.eu/api/v2/record' + recordId + '.json?wskey=[API_KEY]');
```

Comment:
Am I right in thinking that this will never work as-is, due to the absence of an actual API key?

Let's make that configurable by [UI extension configuration parameter](https://www.contentful.com/developers/docs/references/content-management-api/#/reference/ui-extensions/configuration-parameters), in the options property.

Reply:
There is an installation parameter right now, to provide an API key for the record API.